common/functions/src/scalars/maths/log.rs

```diff
@@ -0,0 +1,107 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::f64::consts::E;
+use std::fmt;
+
+use common_datavalues::prelude::*;
+use common_datavalues::DataSchema;
+use common_datavalues::DataType;
+use common_exception::Result;
+
+use crate::scalars::function_factory::FunctionDescription;
+use crate::scalars::function_factory::FunctionFeatures;
+use crate::scalars::Function;
+
+#[derive(Clone)]
+pub struct LogFunction {
+    _display_name: String,
+    default_base: f64,
+}
+
+impl LogFunction {
+    pub fn try_create(_display_name: &str) -> Result<Box<dyn Function>> {
+        Ok(Box::new(Self {
+            _display_name: _display_name.to_string(),
+            default_base: E,
+        }))
+    }
+
+    pub fn desc() -> FunctionDescription {
+        FunctionDescription::creator(Box::new(Self::try_create))
+            .features(FunctionFeatures::default().deterministic())
+    }
+}
+
+impl Function for LogFunction {
+    fn name(&self) -> &str {
+        &*self._display_name
+    }
+
+    fn variadic_arguments(&self) -> Option<(usize, usize)> {
+        Some((1, 2))
+    }
+
+    fn return_type(&self, _args: &[DataType]) -> Result<DataType> {
+        Ok(DataType::Float64)
+    }
+
+    fn nullable(&self, _input_schema: &DataSchema) -> Result<bool> {
+        Ok(true)
+    }
+
+    fn eval(&self, columns: &DataColumnsWithField, _input_rows: usize) -> Result<DataColumn> {
+        let result = if columns.len() == 1 {
+            // Log(num) with default_base if one arg
+            let num_series = columns[0]
+                .column()
+                .to_minimal_array()?
+                .cast_with_type(&DataType::Float64)?;
+            num_series
+                .f64()?
+                .apply_cast_numeric(|v| v.log(self.default_base))
+        } else {
+            // Log(base, num) if two args
+            let num_series = columns[1]
+                .column()
+                .to_minimal_array()?
+                .cast_with_type(&DataType::Float64)?;
+            let base_column: &DataColumn =
+                &columns[0].column().cast_with_type(&DataType::Float64)?;
+            match base_column {
+                DataColumn::Constant(v, _) => {
+                    let base = DFTryFrom::try_from(v.clone())?;
+                    num_series.f64()?.apply_cast_numeric(|v| v.log(base))
+                }
+                DataColumn::Array(base_series) => {
+                    let iter = num_series
+                        .f64()?
+                        .into_no_null_iter()
+                        .zip(base_series.f64()?.into_no_null_iter())
```

Comment:
You can add a test cast:

log(3, null) -> null
log(null, 3) -> null

Reply:
our arrow version do not support casting NULL to a float null array yet, so the stateless test LOG(NULL) would raise an casting error but not the expected a NULL result: https://github.com/jorgecarleitao/arrow2/blob/v0.6.2/src/compute/cast/mod.rs#L260

but in the latest arrow version it is supported: https://github.com/apache/arrow-rs/blob/v0.7.0/arrow/src/compute/kernels/cast.rs#L72

i think after upgrading it to v0.7.0, the NULL behaviour would become expected.